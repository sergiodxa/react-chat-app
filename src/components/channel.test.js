import React from "react";
import { render, fireEvent, getNodeText } from "@testing-library/react";

import Channel from "./channel";

test("Channel component", () => {
  const onClick = jest.fn();

  const { getByLabelText } = render(
    <Channel name="testing" onClick={onClick} />
  );

  const button = getByLabelText("Change to channel testing");

  expect(getNodeText(button)).toBe("# testing");

  fireEvent.click(button);

  expect(onClick).toHaveBeenCalledWith("testing");
});

test("Channel component active", () => {
  const { asFragment } = render(
    <Channel name="general" onClick={jest.fn()} isActive />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("Channel component non- active", () => {
  const { asFragment } = render(<Channel name="general" onClick={jest.fn()} />);
  expect(asFragment()).toMatchSnapshot();
});
