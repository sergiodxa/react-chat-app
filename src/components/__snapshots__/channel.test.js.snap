// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel component active 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: #783CEE;
}

.emotion-0 {
  background: none;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 1rem;
  padding: .5rem;
  outline: none;
  text-align: left;
  width: 100%;
}

<div
    class="emotion-1"
  >
    <button
      aria-label="Change to channel general"
      class="emotion-0"
    >
      # general
    </button>
  </div>
</DocumentFragment>
`;

exports[`Channel component non- active 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: none;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  font-size: 1rem;
  padding: .5rem;
  outline: none;
  text-align: left;
  width: 100%;
}

<div
    class="emotion-1"
  >
    <button
      aria-label="Change to channel general"
      class="emotion-0"
    >
      # general
    </button>
  </div>
</DocumentFragment>
`;
